brave/src/main/java/brave/propagation/ThreadLocalCurrentTraceContext.java

```diff
@@ -42,13 +42,27 @@ public static CurrentTraceContext create() {
     return new Builder().build();
   }
 
-  public static CurrentTraceContext.Builder newBuilder() {
+  public static Builder newBuilder() {
     return new Builder();
   }
 
-  static final class Builder extends CurrentTraceContext.Builder {
+  /**
+   * This component is backed by a possibly static shared thread local. Call this to clear the
+   * reference when you are sure any residual state is due to a leak.
```

Comment:
Is it worth adding something like "This is generally only useful in tests"?

Reply:
sure

Reply:
well it is being done in a hack way in sleuth...

here 
https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java#L123-L126

and also here

https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/scheduling/TraceSchedulingAspect.java#L63-L65

While I can't say either are correct, it might be that some bad instrumentation (ex written by an end user) leaks, and someone wants to correct it.

Reply:
regardless, your statement is correct.. this should (not must) be only used in tests.